plugins/funind/invfun.ml

```diff
@@ -88,102 +74,93 @@ let revert_graph kn post_tac hid g =
    \end{enumerate}
 *)
 
-let functional_inversion kn hid fconst f_correct : Tacmach.tactic =
-  fun g ->
-    let old_ids = List.fold_right Id.Set.add  (pf_ids_of_hyps g) Id.Set.empty in
-    let sigma = project g in
-    let type_of_h = pf_unsafe_type_of g (mkVar hid) in
-    match EConstr.kind sigma type_of_h with
-      | App(eq,args) when EConstr.eq_constr sigma eq (make_eq ())  ->
-          let pre_tac,f_args,res =
-            match EConstr.kind sigma args.(1),EConstr.kind sigma args.(2) with
-              | App(f,f_args),_ when EConstr.eq_constr sigma f fconst ->
-                  ((fun hid -> Proofview.V82.of_tactic (intros_symmetry (Locusops.onHyp hid))),f_args,args.(2))
-              |_,App(f,f_args) when EConstr.eq_constr sigma f fconst ->
-                 ((fun hid -> tclIDTAC),f_args,args.(1))
-              | _ -> (fun hid -> tclFAIL 1 (mt ())),[||],args.(2)
-          in
-          tclTHENLIST [
-            pre_tac hid;
-            Proofview.V82.of_tactic (generalize [applist(f_correct,(Array.to_list f_args)@[res;mkVar hid])]);
-            thin [hid];
-            Proofview.V82.of_tactic (Simple.intro hid);
-            Proofview.V82.of_tactic (Inv.inv Inv.FullInversion None (NamedHyp hid));
-            (fun g ->
-               let new_ids = List.filter (fun id -> not (Id.Set.mem id old_ids)) (pf_ids_of_hyps g) in
-               tclMAP (revert_graph kn pre_tac)  (hid::new_ids)  g
-            );
-          ] g
-      | _ -> tclFAIL 1 (mt ()) g
-
-
-let error msg = user_err Pp.(str msg)
+let functional_inversion kn hid fconst f_correct gl =
```

Comment:
Same remark.

Reply:
Ok, I've amended these two cases not to abstract over gl in the definition.